app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt

```diff
@@ -147,15 +219,49 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
         activity?.apply {
             recycler_view.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
             recycler_view.addItemDecoration(DividerItemDecoration(this,
-                resources.getDimensionPixelSize(R.dimen.divider_item_decorator_bound_start),
-                resources.getDimensionPixelSize(R.dimen.divider_item_decorator_bound_end)))
+                    resources.getDimensionPixelSize(R.dimen.divider_item_decorator_bound_start),
+                    resources.getDimensionPixelSize(R.dimen.divider_item_decorator_bound_end)))
             recycler_view.itemAnimator = DefaultItemAnimator()
             // TODO - use a ViewModel Mapper instead of using settings on the adapter
-            recycler_view.adapter = ChatRoomsAdapter(this,
-                    settingsRepository.get(serverInteractor.get()!!)!!) { chatRoom ->
-                presenter.loadChatRoom(chatRoom)
+
+            val baseAdapter = ChatRoomsAdapter(this,
+                    settingsRepository.get(serverInteractor.get()!!)) { chatRoom -> presenter.loadChatRoom(chatRoom) }
+
+            sectionedAdapter = SimpleSectionedRecyclerViewAdapter(this, R.layout.item_chatroom_header, R.id.text_chatroom_header, baseAdapter!!)
+            recycler_view.adapter = sectionedAdapter
+        }
+    }
+
+    private fun setSections() {
+        //Don't add section if not grouping by RoomType
+        if (!SharedPreferenceHelper.getBoolean(Constants.CHATROOM_GROUP_BY_TYPE_KEY, false)) {
+            sectionedAdapter?.clearSections()
+            return
+        }
+
+        val sections = ArrayList<SimpleSectionedRecyclerViewAdapter.Section>()
+
+        sectionedAdapter?.baseAdapter?.dataSet?.let {
+            var previousChatRoomType = ""
+
+            for ((position, chatRoom) in it.withIndex()) {
+                val type = chatRoom.type.toString()
+                if (type != previousChatRoomType) {
+                    val title = when (type) {
+                        RoomType.CHANNEL.toString() -> "Channels"
+                        RoomType.PRIVATE_GROUP.toString() -> "Private Groups"
+                        RoomType.DIRECT_MESSAGE.toString() -> "Direct Messages"
+                        RoomType.LIVECHAT.toString() -> "Live Chats"
```

Comment:
Pretty sure this doesn't support localization of strings (different languages or if an admin wants to rename "private groups" to "private rooms" but it sounds like you will do that in a later change. Just noting it here.

Reply:
Thanks @bizzbyster!!
I don't know. Is there any option in server to rename Headers?